### PR TITLE
js: Checks that non-preemptible machines are not set as spot machines

### PR DESCRIPTION
**Pull request by HongjianZhang**
Before this commits, the chooseSize function does not distinguish
between spot and reserved machines. It simply assumes that all
machines can be preempted. This is not always the case, e.g. Amazon
t2.micro can only work when using reserved machines. This commit
makes sure that user is either using a spot machine or preemptable
is set to False in blueprint, and warns user to change the config
otherwise.

### CODE REVIEW

File: js/bindings/bindings.js [hunk 1]
```diff
@@ -795,11 +796,18 @@ class Machine {
    * Sets the machine's size attribute to an instance size (e.g., m2.xlarge),
    * based on the Machine's specified provider, region, and hardware. Throws
    * an error if provider is not valid or given machine requirements cannot
-   * be satisfied by any size.
+   * be satisfied by any size. Also throws an error if user requests a spot
```
Comment (kklin) [inline js/bindings/bindings.js]: I think "spot" should be removed here as well. Something like "Also throws an error if the user requests a preemptible instance for a size that cannot be preempted."

File: js/bindings/bindings.js [hunk 2]
```diff
@@ -795,11 +796,18 @@ class Machine {
...
+   * machine for a size that cannot be preempted.
    * @private
    * @returns {void}
    */
   chooseSize() {
+    if (this.preemptible && this.size !== ''
+        && nonPreemptibleSizes[this.provider].includes(this.size)) {
+      throw new Error(`Requested size ${this.size} can not be preemptible.` +
+        ' Please choose a different size or set preemptible to be False in' +
```
Comment (kklin) [inline js/bindings/bindings.js]: I don't think "in blueprint" is necessary. This error message will get thrown when running the blueprint, so it's natural for the user to make the adjustment in the blueprint.

File: js/bindings/bindingsTest.js [hunk 3]
```diff
@@ -240,6 +240,18 @@ describe('Bindings', () => {
         region: 'eu-west-1',
       }]);
     });
+    it('errors if preemptible set to true and requested '
+      + 'size is spot instance for the given provider', () => {
```
Comment (kklin) [inline js/bindings/bindingsTest.js]: Same here re "spot". Also, I think you meant to write "is not a spot instance"?

File: js/bindings/nonPreemptibleSizes.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "Amazon": [
+  	"t2.micro"
+  ]
+}
```
Comment (kklin) [inline js/bindings/nonPreemptibleSizes.json]: This file seems to be missing a newline at the end of the file

File: js/bindings/package.json [hunk 5]
```diff
@@ -7,6 +7,7 @@
     "bindings.js",
     "digitalOceanDescriptions.json",
     "googleDescriptions.json"
```
Comment (kklin) [inline js/bindings/package.json]: Do you need a comma here? Can you try `npm install` on this package and make sure it doesn't error?

**Comment by HongjianZhang** [comment]
Fixed.

**Pull request merged**